src/popup/modules/UserInterface.js

```diff
@@ -435,6 +437,37 @@ function triggerFileSave(file, filename, requestDownloadPermissions) {
     });
 }
 
+/**
+ * If qrcode text was an URL i.e contains :// this function generates a filename using the domain and replacing all special characters with _
+ *.If not return "qrcode"
+ *
+ * @function
+ * @private
+ * @param { } 
+ * @returns {filename}
+ */
+function generateFilename(){
+	var firstIndex = qrCodeInputText.indexOf("://");
+	if(firstIndex > 0) {
+		// URL found
+		var filename = qrCodeInputText.substring(firstIndex+3);
+		var wwwIndex = filename.indexOf("www");
+		if(wwwIndex >= 0){
+			// remove www(*.).
+			filename = filename.substring(filename.indexOf(".")+1);
+		}
+		// keep text until first "/"
+		filename = filename.substring(0,filename.indexOf("/"));
+		// replace . by _
+		//filename = filename.replace(/\./g,"_");
+		// Replace all strange characters
+		filename = filename.replace(/[^a-z0-9_-]/g,"_");
+		// prepend "qrcode"
+		return "qrcode-" + filename;			
+	}
+	return "qrcode";
```

Comment:
Generally, as for the fallback, don't you think it could still make sense to return the first, say 10 chars?

Reply:
What, if they consist of special characters.
Than you have a lot of underscores in the fiename.
The URL syntax is quite restricted, but any free text not.

Reply:
Okay, valid argument.